.github/workflows/embedding-sdk-component-tests.yml

```diff
@@ -0,0 +1,246 @@
+    runs-on: ubuntu-latest
+    outputs:
+      target-ref: ${{ steps.get-target-ref.outputs.value }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Decide ref to checkout
+        id: get-target-ref
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo "value=${{ github.base_ref }}" >> "$GITHUB_OUTPUT"
+          else
+            echo "value=$(git rev-parse HEAD^1)" >> "$GITHUB_OUTPUT"
+          fi
+
+  # This job ensures that the Embedding SDK package from the current commit (N) works with the
+  # Embedding SDK Bundle (Metabase JAR) from the previous commit (N-1).
+  # N - 1 is a target of a pull request, or a literal previous commit when this job is run on push in a branch.
+  sdk-component-tests-backward-compatibility:
+    name: "SDK component tests (Backward compatibility)"
+    needs: [decide-target-ref]
+    runs-on: ubuntu-22.04
+    timeout-minutes: 45
+    env:
+      MB_EDITION: ee
+      DISPLAY: ""
+      MB_RUN_MODE: "e2e"
+      METASTORE_DEV_SERVER_URL: ${{ secrets.METASTORE_DEV_SERVER_URL }}
+      # Any env starting with `CYPRESS_` will be available to all Cypress tests via `Cypress.env()`
+      # Example: you can get `CYPRESS_FOO` with `Cypress.env("FOO")`
+      CYPRESS_MB_ALL_FEATURES_TOKEN: ${{ secrets.STAGING_MB_ALL_FEATURES_TOKEN }}
+      CYPRESS_MB_STARTER_CLOUD_TOKEN: ${{ secrets.STAGING_MB_STARTER_CLOUD_TOKEN }}
+      CYPRESS_MB_PRO_CLOUD_TOKEN: ${{ secrets.STAGING_MB_PRO_CLOUD_TOKEN }}
+      CYPRESS_MB_PRO_SELF_HOSTED_TOKEN: ${{ secrets.STAGING_MB_PRO_SELF_HOSTED_TOKEN }}
+      COMMIT_INFO_MESSAGE: ${{ github.event.pull_request.title || github.event.head_commit.message || github.event.head.sha }}
+      TZ: US/Pacific # to make node match the instance tz
+      CYPRESS_CI: true
+
+    steps:
+      - name: Check out the code from the previous commit
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.decide-target-ref.outputs.target-ref }}
+
+      - name: Prepare back-end environment
+        uses: ./.github/actions/prepare-backend
+
+      - name: Prepare front-end environment
+        uses: ./.github/actions/prepare-frontend
+
+      - name: Attempt to retrieve the ee uberjar artifact from the previous commit
+        id: download-uberjar
+        uses: ./.github/actions/fetch-artifact
+        with:
+          name: metabase-ee-${{ needs.decide-target-ref.outputs.target-ref }}-uberjar
+        continue-on-error: true # do not break the whole job if this step doesn't succeed
+
+      # Build the uberjar from the previous commit only it we couldn't fetch it.
+      # `step.foo.OUTCOME` is impotant because it looks at the step status *before* continue-on-error is applied.
+      # Do not confuse it with `step.foo.conclusion`.
+      - name: Build MB uberjar from the commit N-1
+        if: steps.download-uberjar.outcome != 'success'
+        run: ./bin/build.sh
```

Comment:
Ok, I've confirmed that this step works by not realizing that (in PRs) we will not get the commit as the ref, but the actual branch name.

<img width="516" height="203" alt="Image" src="https://github.com/user-attachments/assets/27647fcd-9a9e-4cda-bbce-dcf861f7254f" />

It attempted to download the jar, it failed, but the job continued just as we expected it to:

<img width="1057" height="328" alt="Image" src="https://github.com/user-attachments/assets/cf3f06d3-b211-4633-bb67-483436b19ecc" />